src/component/EventCardService.js

```diff
@@ -290,53 +290,53 @@ export const fetchEventById = async (eventId) => {
   }
 };
 
-/*export const fetchUserEvents = async (uid) => {
+export const fetchUserSignedUpOutreaches = async () => {
   try {
-    const userQuery = query(
-      collection(db, USERS_COLLECTION),
-      where("uid", "==", uid)
-    );
-    const userDocRef = await getDocs(userQuery);
-    if (userDocRef.docs.length === 0) {
-      console.error("User document not found for uid:", uid);
-      return [];
-    }
-    // const userDocID = userDocRef.docs[0].id;
-    const userData = userDocRef.docs[0].data();
-    // reference for the userdoc
-    // const userRef = doc(db, USERS_COLLECTION, userDocID);
-    // const userDoc = await getDoc(userRef);
-
-    // if (!userDoc.exists()) {
-    //   console.error("User not found:", uid);
-    //   return [];
-    // }
+    const fAuth = getAuth();
+    let userSignedUpOutreaches = [];
 
-    // const eventIds = userDoc.data().outreachEvents || [];
-    const eventIds = userData.outreachEvents || [];
-    const eventsData = [];
-
-    for (let eventId of eventIds) {
-      const eventRef = doc(db, OUTREACH_EVENTS_COLLECTION, eventId);
-      const eventDoc = await getDoc(eventRef);
-      if (eventDoc.exists()) {
-        const eventData = eventDoc.data();
-        eventsData.push({
-          ...eventData,
-          id: eventDoc.id,
-        });
+    onAuthStateChanged(fAuth, async (user) => {
```

Comment:
Accept uid from the function parameter and not onAuthStateChanged

Reply:
Updated the function with uid.